include/drivers/dac.h

```diff
@@ -0,0 +1,161 @@
+#endif
+
+/**
+ * @brief DAC driver APIs
+ * @defgroup dac_interface DAC driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/** @brief DAC references. */
+enum dac_reference {
+	DAC_REF_DEFAULT,   /**< Default, reference cannot be selected. */
+	DAC_REF_VDD_1,     /**< VDD. */
+	DAC_REF_VDD_1_2,   /**< VDD/2. */
+	DAC_REF_VDD_1_3,   /**< VDD/3. */
+	DAC_REF_VDD_1_4,   /**< VDD/4. */
+	DAC_REF_INTERNAL,  /**< Internal. */
+	DAC_REF_EXTERNAL0, /**< External, input 0. */
+	DAC_REF_EXTERNAL1, /**< External, input 1. */
+};
+
+/**
+ * @brief Structure for specifying the configuration of a DAC channel.
+ */
+struct dac_channel_cfg {
+	/**
+	 * Channel identifier of the DAC that should be configured.
+	 */
+	u8_t channel_id;
+
+	/**
+	 * Reference selection.
+	 */
+	enum dac_reference reference;
+
+	/**
+	 * Enable buffered output if available in underlying hardware.
+	 */
+	bool buffered;
+};
+
+/**
+ * @brief Type definition of DAC API function for configuring a channel.
+ * See dac_channel_setup() for argument descriptions.
+ */
+typedef int (*dac_api_channel_setup)(struct device *dev,
+				     const struct dac_channel_cfg *channel_cfg);
+
+/**
+ * @brief Type definition of DAC API function for setting a write request.
+ * See dac_write() for argument descriptions.
+ */
+typedef int (*dac_api_write)(struct device *dev,
+			    u8_t channel, u32_t value, u8_t resolution);
+
+/**
+ * @brief DAC driver API
+ *
+ * This is the mandatory API any DAC driver needs to expose.
+ */
+struct dac_driver_api {
+	dac_api_channel_setup channel_setup;
+	dac_api_write         write;
+	u16_t ref_internal;	/* mV */
+};
+
+/**
+ * @brief Configure a DAC channel.
+ *
+ * It is required to call this function and configure each channel before it is
+ * selected for a write request.
+ *
+ * @param dev          Pointer to the device structure for the driver instance.
+ * @param channel_cfg  Channel configuration.
+ *
+ * @retval 0       On success.
+ * @retval -EINVAL If a parameter with an invalid value has been provided.
+ */
+__syscall int dac_channel_setup(struct device *dev,
+				const struct dac_channel_cfg *channel_cfg);
+
+static inline int z_impl_dac_channel_setup(struct device *dev,
+				const struct dac_channel_cfg *channel_cfg)
+{
+	const struct dac_driver_api *api =
+				(const struct dac_driver_api *)dev->driver_api;
+
+	return api->channel_setup(dev, channel_cfg);
+}
+
+/**
+ * @brief Set a write request.
+ *
+ * @param dev         Pointer to the device structure for the driver instance.
+ * @param channel     Number of the channel to be used.
+ * @param value       Data to be written to DAC output registers.
+ * @param resolution  Desired resolution of the DAC (depends on device
+ *                    capabilities).
+ *
+ * @retval 0        On success.
+ * @retval -EINVAL  If a parameter with an invalid value has been provided.
+ * @retval -ENOTSUP If the requested resolution is not supported.
+ */
+__syscall int dac_write(struct device *dev,
```

Comment:
I propose this to be `dac_write_value` or `dac_write_single` so that `dac_write` is reserved for sequences.